src/bitbucket-pr.coffee

```diff
@@ -15,243 +11,267 @@
 # Author:
 #   tshedor
 
-url = require('url')
-querystring = require('querystring')
+DEFAULT_ROOM = process.env.HUBOT_BITBUCKET_PULLREQUEST_ROOM
+
+# Set default actions to announce
+DEFAULT_ANNOUNCE_OPTIONS = [
+  'created', 'updated', 'declined', 'merged',
+  'comment_created', 'approve', 'unapprove'
+]
+
+getEnvAnnounceOptions = ->
+  # Replace announce options if set in the environment
+  if process.env.HUBOT_BITBUCKET_PULLREQUEST_ANNOUNCE
+    process.env.HUBOT_BITBUCKET_PULLREQUEST_ANNOUNCE.replace(/[^a-z\,]+/, '').split(',')
+
+ANNOUNCE_OPTIONS = getEnvAnnounceOptions() ? DEFAULT_ANNOUNCE_OPTIONS
+
+ENCOURAGEMENTS = [
+  ':thumbsup:', 'That was a nice thing you did.', 'Boomtown',
+  'BOOM', 'Finally.', 'And another request bites the dust.'
+]
+
+encourageMe = ->
+  ENCOURAGEMENTS[Math.floor(Math.random() * ENCOURAGEMENTS.length)]
+
+class PullRequestEvent
+  constructor: (@robot, @resp, @type) ->
+    @actor = @resp.actor.display_name
+    @title = @resp.pullrequest.title
+    @source_branch = @resp.pullrequest.source.branch.name
+    @destination_branch = @resp.pullrequest.destination.branch.name
+    @repo_name = @resp.repository.name
+    @pr_link = @resp.pullrequest.links.html.href
+    @reason = if @resp.reason isnt '' then ":\n\"#{@resp.pullrequest.reason}\"" else "."
+
+  getReviewers: ->
+    if @resp.pullrequest.reviewers.length > 0
+      reviewer_names = for reviewer in @resp.pullrequest.reviewers
+        "#{reviewer.display_name}"
+      reviewers_names.join(", ")
+    else
+      'no one in particular'
 
-module.exports = (robot) ->
-  robot.router.post '/hubot/bitbucket-pr', (req, res) ->
-    query = querystring.parse(url.parse(req.url).query)
+  branchAction: (action_name, action_desc) ->
+    "#{@actor} *#{action_name}* pull request \"#{@title},\" #{action_desc}
+    `#{@source_branch}` and `#{@destination_branch}` into a `#{@repo_name}`
+    super branch#{@reason}"
 
-    # Set default actions to announce
-    announce_options = ['created', 'updated', 'declined', 'merged', 'comment_created', 'approve', 'unapprove']
+  getMessage: ->
+    msg = switch
```

Comment:
The `msg` variable is declared here but `msg` isn't explicitly returned. Again, I'm not terribly familiar with Coffeescript, so I assume this is an implied `return`, but I'd prefer to either have `msg` returned after the `switch` or to not declare it at all.